lib/cosmos/tools/cmd_tlm_server/cmd_tlm_server.rb

```diff
@@ -240,7 +240,16 @@ def initialize(
     # Map any targets without interfaces to the dummy replay interface.
     # Targets will only have an interface already mapped if the replay_routers
     # flag was passed to the server.
-    def map_targets_to_interfaces
+    def replay_map_targets_to_interfaces
+      # Try to map existing interfaces to targets
+      if @interfaces
+        @interfaces.all.each do |name, interface|
+          interface.target_names.each do |target|
+            System.targets[target] = interface
```

Comment:
This is overwriting the target with an interface.   It should set the target's interface.